docs/resources/cluster_v2.md

```diff
@@ -580,38 +353,193 @@ EOF
-      control_plane_role = true
-      etcd_role = true
-      worker_role = true
-      quantity = 1
-      machine_config {
-        kind = rancher2_machine_config_v2.foo.kind
-        name = rancher2_machine_config_v2.foo.name
+    # ...
+  }
+}
+```
+
+
+### Customize the cluster agent and the fleet agent
+
+This argument is available in Rancher v2.7.5 and above.
+
+You can configure the tolerations, affinity rules, and resource requirements for the `cattle-cluster-agent` and `fleet-agent` deployments.
+
+The example below demonstrates how to set `cluster_agent_deployment_customization` and `fleet_agent_deployment_customization`:
+
+```hcl
+resource "rancher2_cluster_v2" "foo" {
+  name = "foo"
+  kubernetes_version = "rke2/k3s-version"
+  cluster_agent_deployment_customization {
+    append_tolerations {
+      key    = "tolerate/control-plane"
+      effect = "NoSchedule"
+      value  = "true"
+    }
+    # Each toleration must be passed separately
+    append_tolerations {
+      key    = "tolerate/etcd"
+      effect = "NoSchedule"
+      value  = "true"
+    }
+    # Override_affinity accepts a string whose value is an Affinity, a k8s native object, in JSON format
+    override_affinity = <<EOF
+{
+  "nodeAffinity": {
+    "requiredDuringSchedulingIgnoredDuringExecution": {
+      "nodeSelectorTerms": [{
+        "matchExpressions": [{
+          "key": "not.this/nodepool",
+          "operator": "In",
+          "values": [
+            "true"
+          ]
+        }]
+      }]
+    }
+  }
+}
+EOF
+    # The resource units follow Kubernetes' standard
+    override_resource_requirements {
+      cpu_limit      = "800"
+      cpu_request    = "500"
+      memory_limit   = "800"
+      memory_request = "500"
+    }
+  }
+  
+  fleet_agent_deployment_customization {
+    # The same format and requirement for values as the cluster_agent_deployment_customization argument
+    # ...
+  }
+  rke_config {
+    # In the case of a node-driver cluster
+    machine_pools {
+      # ...
       }
+    } 
+}
+```
+
+### Enable Pod Security Policy Admission Configuration Template (PSACT) on the cluster
+
+This argument is available in Rancher v2.7.2 and above.
+
+**Note:** When PSACT is enabled in an RKE2 or K3s cluster, Rancher webhook sets the `kube-apiserver-arg` to the Pod Security Admission mount path. To suppress Terraform constantly trying to reconcile that arg, it must be set locally.
+
+```hcl
+locals {
+  version = "rke2"  // "k3s" for K3s clusters
+  rancher_psact_mount_path = "/etc/rancher/${local.version}/config/rancher-psact.yaml"
+  kube_apiserver_arg = var.default_psa_template != null && var.default_psa_template != "" ? ["admission-control-config-file=${local.rancher_psact_mount_path}"] : []
+}
+
+resource "rancher2_cluster_v2" "foo" {
+  name = "foo"
+  kubernetes_version = "rke2/k3s-version"
+  enable_network_policy = false
+  // There are two builtin PSACT: rancher-privileged and rancher-restricted. You can also create new ones.
```

Comment:
It might be helpful to link to the psact docs somewhere so the difference between `rancher-privileged` and `rancher-restricted` is clear ([link to docs](https://ranchermanager.docs.rancher.com/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/psa-config-templates))

Reply:
The link has been provided in the detailed description of the argument `default_pod_security_admission_configuration_template_name` on this page.

Reply:
Nice, I must have missed it